### ISSUE TITLE
ZSTD instead of XZ?

### ISSUE THREAD

**Issue by M-Gonzalo** [burrito-elixir/burrito#130]
Hi! This is a great, great project!

I noticed Burrito is using xz to compress its payload (lzma2 I'm guessing). It's certainly a good enough algorithm/format, with good compression and nice decompression speed, and it can be extracted basically anywhere.

There has been a shift though, in the last years, towards zstd for the same use cases one might otherwise use xz. There are many examples, one of them being the distribution format of Arch Linux packages, or squashfs images on live ISOs. 

The reason/s is mainly that zstd allows for ~10x decompression speed while maintaining a competitive ratio. And, depending on the original uncompressed size, it can deliver a much better compression, using the --long option (built-in long-range deduplication), which allows it to "see" orders of magnitude more information and include it in its LZ dictionary.

The possible downside could be a slightly reduced presence of the decompressor program (especially in very old, outdated systems)

In summary, zstd will speed up the start time of a Burrito app, and probably reduce considerably its size, at the cost of a (possibly) reduced target count.

**Comment by mmower** [comment]
Certainly sounds like it would make a good option, where you don't think distribution to older systems is an issue.

**Comment by doawoo** [comment]
It looks like zig HAS a zstd decompressor already built in!
(https://github.com/ziglang/zig/pull/14394)

Seems like they don't have a compressor yet oddly enough. But this is something we could work with

**Comment by doawoo** [comment]
After doing some light testing I can see that zstd produces a much larger packed binary... even in the small `cli_example` app when using the very CPU heavy/aggressive compression level:

<img width="563" alt="image" src="https://github.com/user-attachments/assets/d276645b-8615-4f7e-8f6f-531e157d8fcc">

The unpack/first run times:

```
./example_cli_app_macos_m1  3.45s user 0.43s system 89% cpu 4.357 total
./example_cli_app_macos_m1_zstd  5.90s user 0.44s system 89% cpu 7.065 total
```

It even seems slower during unpack, but that might also be due to Zig's decompressor implementation.

Ultimately I tend prefer the binary be smaller, so for now, I think we'll be sticking with XZ.
